packages/server/src/assertions/bundler-assertions.ts

```diff
@@ -0,0 +1,30 @@
+import { BundlerTypes } from '../orchestrator/state';
+
+export class BundlerAssertionError extends Error {
+  constructor(message: string) {
+    super(`INTERNAL ASSERTION FAILURE:
+    While bundling test files, ${message} was received.
+    this is almost certainly a bug in BigTest, and we would be very grateful if you would report
+    it at https://github.com/thefrontside/bigtest/issues/new
+    `)
+
+    Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
🤔 not grokking this 🤔

Reply:
The problem is that Javascript's built-in class Error breaks the prototype chain by switching the object to be constructed (i.e. this) to a new, different object, when you call super and that new object doesn't have the expected prototype chain, i.e. it's an instance of Error not of CustomError.

```typescript
class CustomError extends Error {
  constructor(message?: string) {
    super(message); // 'Error' breaks prototype chain here
    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
  }
}
```
[docs]( https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html)

Using new.target has the advantage that you don't have to hardcode the prototype